src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/KestrelEventSource.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics.Tracing;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.Http;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Internal.Infrastructure
+{
+    [EventSource(Name = "Microsoft-AspNetCore-Server-Kestrel")]
+    public sealed class KestrelEventSource : EventSource
+    {
+        public static readonly KestrelEventSource Log = new KestrelEventSource();
+
+        private KestrelEventSource()
+        {
+        }
+
+        [NonEvent]
+        public void ConnectionStart(Connection connection)
+        {
+            if (IsEnabled())
+            {
+                ConnectionStart(
+                    connection.ConnectionId,
+                    connection.ListenerContext.ListenOptions.Scheme,
+                    connection.LocalEndPoint.ToString(),
+                    connection.RemoteEndPoint.ToString());
+            }
+        }
+
+        [NonEvent]
+        public void ConnectionStop(Connection connection)
+        {
+            if (IsEnabled())
+            {
+                ConnectionStop(connection.ConnectionId);
+            }
+        }
+
+        [Event(1, Level = EventLevel.Informational)]
```

Comment:
Thoughts @shirhatti? I don't have enough context to know if `EventLevel.Verbose` is more appropriate than `EventLevel.Informational`. This entirely depends on the desired UX.